src/o365/spo/commands/app/app-retract.ts

```diff
@@ -193,7 +152,7 @@ class AppDeployCommand extends SpoCommand {
       }
 
       if (args.options.appCatalogUrl) {
-        return SpoCommand.isValidSharePointUrl(args.options.appCatalogUrl);
+        return SpoAppBaseCommand.isValidSharePointUrl(args.options.appCatalogUrl);
```

Comment:
Since `isValidSharePointUrl` is defined in the `SpoCommand` class, it would be clearer to use that class instead of `SpoAppBaseCommand`